ArduPlane/ArduPlane.cpp

```diff
@@ -499,8 +499,31 @@ void Plane::set_flight_stage(AP_FixedWing::FlightStage fs)
     landing.handle_flight_stage_change(fs == AP_FixedWing::FlightStage::LAND);
 
     if (fs == AP_FixedWing::FlightStage::ABORT_LANDING) {
-        gcs().send_text(MAV_SEVERITY_NOTICE, "Landing aborted, climbing to %dm",
-                        int(auto_state.takeoff_altitude_rel_cm/100));
+        const int32_t& climb_target_agl_cm = auto_state.takeoff_altitude_rel_cm; // this is the LAND waypoint's param1
+        int32_t climb_delta_cm;
+        if (have_rangefinder_correction()) {
+            // if the rangefinder is tracking the ground, we use it to convert climb_target_agl_cm to an altitude delta from our current altitude
+            int32_t height_above_ground_cm = rangefinder_state.height_estimate*100;
+            
+            climb_delta_cm = climb_target_agl_cm - height_above_ground_cm;
+        } else {
+            // if the rangefinder is not tracking the ground, we assume that the LAND waypoint is on the ground
+            // to convert climb_target_agl_cm to an altitude delta from our current altitude
+            int32_t land_wp_alt_rel_cm;
+            if (next_WP_loc.get_alt_cm(Location::AltFrame::ABOVE_HOME, land_wp_alt_rel_cm)) {
```

Comment:
doesn't look right, double check?

Reply:
Looks right to me.

Reply:
Actually, that wp could be in any frame and that may give weird results. Maybe don't convert the frame, but use it only if it's already in relative home..

Reply:
Should be fixed in latest. @magicrub we need to convert the frame, because the land waypoint is supposed to be on the ground. Doesn't matter what frame it is in, it designates the ground, so we convert it to relative and then use it as the ground height.

Reply:
well.. normally yes. However, what about land tuning where we do it up in the air like 20m?